src/rateEngine/RateComponentsFactory.ts

```diff
@@ -0,0 +1,112 @@
+import RateComponent from './RateComponent';
+import AnnualDemand from './billingDeterminants/AnnualDemand';
+import BlockedTiersInDays from './billingDeterminants/BlockedTiersInDays';
+import BlockedTiersInMonths from './billingDeterminants/BlockedTiersInMonths';
+import DemandPerDay from './billingDeterminants/DemandPerDay';
+import DemandTiersInMonths from './billingDeterminants/DemandTiersInMonths';
+import DemandTimeOfUse from './billingDeterminants/DemandTimeOfUse';
+import EnergyTimeOfUse from './billingDeterminants/EnergyTimeOfUse';
+import FixedPerDay from './billingDeterminants/FixedPerDay';
+import FixedPerMonth from './billingDeterminants/FixedPerMonth';
+import HourlyEnergy from './billingDeterminants/HourlyEnergy';
+import MonthlyDemand from './billingDeterminants/MonthlyDemand';
+import MonthlyEnergy from './billingDeterminants/MonthlyEnergy';
+import SurchargeAsPercent from './billingDeterminants/SurchargeAsPercent';
+import LoadProfile from './LoadProfile';
+import PriceProfile from './PriceProfile';
+import RateElement from './RateElement';
+import type { RateElementInterface } from './types';
+
+export default class RateComponentsFactory {
```

Comment:
I don't think the BillingDeterminantsFactory should be an implicit part of the RateComponents factory.

They have very distinct jobs... selecting strategies for very different purposes. (generating rate component data vs. an algorithm to calculate "billing determinants")

I assume that RateComponentsFactory feels messy to you because it is awkwardly shoehorned into the `RateElement.js` file. I agree with that. What if instead of combining them, we:

1. Leave BillingDeterminantsFactory.js in place (with any additional changes you made to it)
2. Create a RateComponentsFactory.js file
3. Pull the rate components strategies out of the factory into a dedicated folder (just like the billing determinant strategies). There should be a "default" rate component strategy, which is basically "nothing to see here!"